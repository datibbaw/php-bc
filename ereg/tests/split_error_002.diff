004+ array(7) {
005+   [0]=>
006+   string(0) ""
007+   [1]=>
008+   string(1) "h"
009+   [2]=>
010+   string(1) "e"
011+   [3]=>
012+   string(1) "l"
013+   [4]=>
014+   string(1) "l"
015+   [5]=>
016+   string(1) "o"
017+   [6]=>
018+   string(0) ""
019+ }
004- 
005- Warning: split(): REG_EMPTY in %s on line %d
006- bool(false)
007- 
008- Deprecated: Function split() is deprecated in %s on line %d
009- 
010- Warning: split(): REG_EPAREN in %s on line %d
011- bool(false)
015- Warning: split(): REG_EBRACK in %s on line %d
020- Warning: split(): REG_EPAREN in %s on line %d
023+ Warning: preg_split(): Compilation failed: missing ) at offset 3 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
025- Warning: split(): REG_BADRPT in %s on line %d
028+ Warning: preg_split(): Compilation failed: missing terminating ] for character class at offset 3 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
030- Warning: split(): REG_BADRPT in %s on line %d
033+ Warning: preg_split(): Compilation failed: missing ) at offset 3 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
035- Warning: split(): REG_BADRPT in %s on line %d
038+ Warning: preg_split(): Compilation failed: nothing to repeat at offset 0 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
040- Warning: split() expects parameter 3 to be long, string given in %s on line %d
041- NULL
043+ Warning: preg_split(): Compilation failed: nothing to repeat at offset 0 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
044- 
045- Warning: split(): REG_BADBR in %s on line %d
046- bool(false)
048+ Warning: preg_split(): Compilation failed: nothing to repeat at offset 0 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
049+ bool(false)
049- 
050- Warning: split(): REG_EMPTY in %s on line %d
051- bool(false)
052- 
053- Deprecated: Function split() is deprecated in %s on line %d
054- 
055- Warning: split(): REG_EMPTY in %s on line %d
056- bool(false)
052+ array(1) {
053+   [0]=>
054+   string(5) "hello"
055+ }
056+ 
057+ Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
058+ array(1) {
059+   [0]=>
060+   string(5) "hello"
061+ }
059- 
060- Warning: split(): REG_BADBR in %s on line %d
061- bool(false)
062- 
063- Deprecated: Function split() is deprecated in %s on line %d
064- 
065- Warning: split(): REG_ERANGE in %s on line %d
066- bool(false)
064+ array(7) {
065+   [0]=>
066+   string(0) ""
067+   [1]=>
068+   string(0) ""
069+   [2]=>
070+   string(1) "e"
071+   [3]=>
072+   string(1) "l"
073+   [4]=>
074+   string(1) "l"
075+   [5]=>
076+   string(1) "o"
077+   [6]=>
078+   string(0) ""
079+ }
070- Warning: split(): REG_EESCAPE in %s on line %d
075- Warning: split() expects parameter 3 to be long, string given in %s on line %d
076- NULL
082+ array(7) {
083+   [0]=>
084+   string(0) ""
085+   [1]=>
086+   string(1) "h"
087+   [2]=>
088+   string(1) "e"
089+   [3]=>
090+   string(1) "l"
091+   [4]=>
092+   string(1) "l"
093+   [5]=>
094+   string(1) "o"
095+   [6]=>
096+   string(0) ""
097+ }
101+ Warning: preg_split(): Compilation failed: numbers out of order in {} quantifier at offset 5 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
105+ array(1) {
106+   [0]=>
107+   string(1) "d"
108+ }
110+ Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
111+ 
112+ Warning: preg_split(): No ending delimiter '/' found in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
113+ bool(false)
114+ 
115+ Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
116+ array(1) {
117+   [0]=>
118+   string(1) "1"
119+ }