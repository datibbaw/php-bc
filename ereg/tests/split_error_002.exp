*** Testing split() : error conditions ***

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_EMPTY in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_EPAREN in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_EBRACK in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_EPAREN in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_BADRPT in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_BADRPT in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_BADRPT in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split() expects parameter 3 to be long, string given in %s on line %d
NULL

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_BADBR in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_EMPTY in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_EMPTY in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_BADBR in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_ERANGE in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split(): REG_EESCAPE in %s on line %d
bool(false)

Deprecated: Function split() is deprecated in %s on line %d

Warning: split() expects parameter 3 to be long, string given in %s on line %d
NULL
string(8) "original"
Done