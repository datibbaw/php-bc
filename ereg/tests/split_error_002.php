<?php
/* Prototype  : proto array split(string pattern, string string [, int limit])
 * Description: Split string into array by regular expression 
 * Source code: ext/standard/reg.c
 * Alias to functions: 
 */

/*
 * Test bad regular expressions
 */

echo "*** Testing split() : error conditions ***\n";

$regs = 'original';

var_dump(split("", "hello"));
var_dump(split("c(d", "hello"));
var_dump(split("a[b", "hello"));
var_dump(split("c(d", "hello"));
var_dump(split("*", "hello"));
var_dump(split("+", "hello"));
var_dump(split("?", "hello"));
var_dump(split("(+?*)", "hello", $regs));
var_dump(split("h{256}", "hello"));
var_dump(split("h|", "hello"));
var_dump(split("h{0}", "hello"));
var_dump(split("h{2,1}", "hello"));
var_dump(split('[a-c-e]', 'd'));
var_dump(split('\\', 'x'));
var_dump(split('([9-0])', '1', $regs));

//ensure $regs unchanged
var_dump($regs);

echo "Done";
?>
