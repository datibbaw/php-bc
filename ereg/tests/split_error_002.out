*** Testing split() : error conditions ***

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
array(7) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "h"
  [2]=>
  string(1) "e"
  [3]=>
  string(1) "l"
  [4]=>
  string(1) "l"
  [5]=>
  string(1) "o"
  [6]=>
  string(0) ""
}

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179

Warning: preg_split(): Compilation failed: missing ) at offset 3 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
bool(false)

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179

Warning: preg_split(): Compilation failed: missing terminating ] for character class at offset 3 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
bool(false)

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179

Warning: preg_split(): Compilation failed: missing ) at offset 3 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
bool(false)

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179

Warning: preg_split(): Compilation failed: nothing to repeat at offset 0 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
bool(false)

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179

Warning: preg_split(): Compilation failed: nothing to repeat at offset 0 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
bool(false)

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179

Warning: preg_split(): Compilation failed: nothing to repeat at offset 0 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
bool(false)

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
array(1) {
  [0]=>
  string(5) "hello"
}

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
array(1) {
  [0]=>
  string(5) "hello"
}

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
array(7) {
  [0]=>
  string(0) ""
  [1]=>
  string(0) ""
  [2]=>
  string(1) "e"
  [3]=>
  string(1) "l"
  [4]=>
  string(1) "l"
  [5]=>
  string(1) "o"
  [6]=>
  string(0) ""
}

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
array(7) {
  [0]=>
  string(0) ""
  [1]=>
  string(1) "h"
  [2]=>
  string(1) "e"
  [3]=>
  string(1) "l"
  [4]=>
  string(1) "l"
  [5]=>
  string(1) "o"
  [6]=>
  string(0) ""
}

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179

Warning: preg_split(): Compilation failed: numbers out of order in {} quantifier at offset 5 in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
bool(false)

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
array(1) {
  [0]=>
  string(1) "d"
}

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179

Warning: preg_split(): No ending delimiter '/' found in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 185
bool(false)

Deprecated: Function split() is deprecated in /Users/tjerk/Documents/php-bc/ereg/ereg.php on line 179
array(1) {
  [0]=>
  string(1) "1"
}
string(8) "original"
Done